src/diff/index.js

```diff
@@ -67,14 +67,14 @@ export function diff(
 
 	// When passing through createElement it assigns the object
 	// constructor as undefined. This to prevent JSON-injection.
-	if (newVNode.constructor !== undefined) return null;
+	if (newVNode._node && newVNode._node.constructor !== undefined) return null;
```

Comment:
When we have a function child `_node` won't be defined

Reply:
If it's easier, this could be moved into `createBackingNode()` since it'll still be invoked from within the diff. It just needs to break, doesn't really matter how it breaks.